### PR TITLE
Move checkstyle from maven goal to gradle #156

### PR DESCRIPTION
**Pull request by pg301**
Move checkstyle from maven goal to gradle #156

### CODE REVIEW

**Comment by pg301** [comment]
WIP

File: mobile/build.gradle [hunk 1]
```diff
@@ -122,6 +123,17 @@ def readProguardFiles() {
     return proguardFiles
 }
 
+tasks.withType(Checkstyle) {
+    configFile = rootProject.file('config/checkstyle/checkstyle.xml')
```
Comment (KeenWarrior) [inline mobile/build.gradle]: It will be better to have `checkstyle.xml` under `misc/`

File: mobile/build.gradle [hunk 2]
```diff
@@ -122,6 +123,17 @@ def readProguardFiles() {
...
+
+    source 'src'
+    include '**/*.java'
```
Comment (KeenWarrior) [inline mobile/build.gradle]: also include `**/*.json`, `**/*.xml`

File: mobile/build.gradle (hunk 1 above)
Comment (pg301) [reply mobile/build.gradle]: @garg-anuj Gradle guidelines say that checkstyle.xml should be in a folder called config. I can transfer the file if you want.

Comment (KeenWarrior) [reply mobile/build.gradle]: @pg301 When it is at `config/checkstyle/checkstyle.xml` then it is not needed to be specified. That is the magic of convention. But creating multiple folders for such minimal things will also be wrong way. So please move them to `misc/` :wink:

**Comment by pg301** [comment]
@garg-anuj @danielpassos Some more changes required?

**Comment by KeenWarrior** [comment]
@pg301 I will test and let you know

**Review by KeenWarrior — APPROVED** [review]

**Pull request merged**